lib/User.ts

```diff
@@ -1,6 +1,36 @@
-namespace MSAL {
-    export interface User {
-        username: string;
-        profile: any;
+"use strict";
+
+namespace Msal {
+
+    export class User {
+
+        displayableId: string;
+        name: string;
+        identityProvider: string;
+        userIdentifier:string;
+
+        constructor(displayableId: string, name: string, identityProvider: string, userIdentifier: string) {
+            this.displayableId = displayableId;
+            this.name = name;
+            this.identityProvider = identityProvider;
+            this.userIdentifier = userIdentifier;
+        }
+
+        static createUser(idToken: IdToken, clientInfo: ClientInfo, authority: string): User {
+            let uid: string;
+            let utid: string;
+            if (!clientInfo) {
+                uid = "";
+                utid = "";
+            }
+            else {
+                uid = clientInfo.uid;
+                utid = clientInfo.utid;
+            }
+			
+            authority = authority.replace("common", idToken.tenantId).replace("organizations", idToken.tenantId);
```

Comment:
>authority [](start = 12, length = 9)

if my authority is [https://login.microsoftonline.com/myorganizations.onmicrosoft.com/](https://login.microsoftonline.com/myorganizations.onmicrosoft.com/) then it will break. You need to only check first path in the URL. you code should compare "myorganizations.onmicrosoft.com" with common/organizations..